experimental/packages/web-common/src/SessionSpanProcessor.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Context } from '@opentelemetry/api';
+import {
+  SpanProcessor,
+  Span,
+  ReadableSpan,
+} from '@opentelemetry/sdk-trace-base';
+import { ATTR_SESSION_ID } from '@opentelemetry/semantic-conventions/incubating';
+import { SessionProvider } from './types/SessionProvider';
+
+/**
+ * SessionSpanProcessor is a {@link SpanProcessor} that adds the session.id attribute
+ */
+export class SessionSpanProcessor implements SpanProcessor {
```

Comment:
Q: do we want to have a `SessionSpanProcessor` class exported or would a factory function work too? :thinking: 
When we export the class users will be able to extend it and the type . I wonder if a factory function (`function createSessionSpanProcessor(sessionIdProvider: SessionProvider): SpanProcessor`) would be sufficient in this case. 

It would be somewhat inconsistent with the current other parts of the SDK, but it reduces the public API quite a bit and makes future maintenance easier, as private fields then not part of the public types API.

Reply:
I don't see any downside to exporting a factory function instead, will make the change.